api/storage/elasticsearch/elasticsearch.go

```diff
@@ -85,24 +90,31 @@ func NewStorageManager(conf config.ElasticsearchConfig) (*StorageManager, error)
 	}
 
 	storageManager := &StorageManager{
-		client:       esclient,
-		defaultIndex: conf.Index,
+		client: esclient,
 	}
-	storageManager.createIndex(conf.Index)
+
+	storageManager.setCurrentIndexDay(time.Now().In(time.UTC))
+	go func() {
+		for {
+			now := time.Now().In(time.UTC)
+			storageManager.checkCurrentIndexDay(now)
```

Comment:
I would move the sleep/wait functions here or change the function name to represent it.

Reply:
sound good. 
changed. the function returns the duration time and the duration sleep moved to the for loop